app/controllers/account/trades_controller.rb

```diff
@@ -34,7 +34,11 @@ def update
   end
 
   def securities
-    @pagy, @securities = pagy(Security.order(:name).search(params[:q]), limit: 20)
+    query = params[:q]
+    return render json: [] if query.blank? || query.length < 2 || query.length > 100
+
+    synth_client = Provider::Synth.new(ENV["SYNTH_API_KEY"])
+    @securities = synth_client.search_securities(query:, dataset: "limited", country_code: Current.family.country).securities
```

Comment:
The only change with the provider here is we'll need to set the key via `Setting.synth_api_key` (as we're doing in `Providable`) for self hosting to work.

The diff below should make that work _and_ the combobox display work (**note:** I renamed `_tickers.turbo_stream.erb` to `_security.turbo_stream.erb`)

```diff
diff --git a/app/controllers/account/trades_controller.rb b/app/controllers/account/trades_controller.rb
index 8d31d3a..04f2048 100644
--- a/app/controllers/account/trades_controller.rb
+++ b/app/controllers/account/trades_controller.rb
@@ -37,8 +37,17 @@ class Account::TradesController < ApplicationController
     query = params[:q]
     return render json: [] if query.blank? || query.length < 2 || query.length > 100
 
-    synth_client = Provider::Synth.new(ENV["SYNTH_API_KEY"])
-    @securities = synth_client.search_securities(query:, dataset: "limited", country_code: Current.family.country).securities
+    @securities = Security.security_prices_provider.search_securities(query:, dataset: "limited", country_code: Current.family.country).securities.map do |security|
+      new_security = Security.new(
+        ticker: security[:symbol],
+        name: security[:name],
+        exchange_acronym: security[:exchange_acronym]
+      )
+
+      new_security.define_singleton_method(:logo_url) { security[:logo_url] }
+
+      new_security
+    end
   end
 
   private
diff --git a/app/models/security.rb b/app/models/security.rb
index 7192239..4558b85 100644
--- a/app/models/security.rb
+++ b/app/models/security.rb
@@ -1,4 +1,6 @@
 class Security < ApplicationRecord
+  include Providable
+
   before_save :upcase_ticker
 
   has_many :trades, dependent: :nullify, class_name: "Account::Trade"
@@ -17,7 +19,6 @@ class Security < ApplicationRecord
     "#{ticker} - #{name} (#{exchange_acronym})"
   end
 
-
   private
 
     def upcase_ticker
diff --git a/app/views/account/trades/_tickers.turbo_stream.erb b/app/views/account/trades/_tickers.turbo_stream.erb
deleted file mode 100644
-</div>
\ No newline at end of file
diff --git a/app/views/account/trades/securities.turbo_stream.erb b/app/views/account/trades/securities.turbo_stream.erb
index f7fcaf2..a3128c7 100644
--- a/app/views/account/trades/securities.turbo_stream.erb
+++ b/app/views/account/trades/securities.turbo_stream.erb
@@ -1,7 +1,2 @@
-<%= async_combobox_options @securities.map { |security| 
-      { 
-        display: security,
-        value: security[:symbol]
-      }
-    },
-    render_in: { partial: "account/trades/tickers" } %>
\ No newline at end of file
+<%= async_combobox_options @securities,
+    render_in: { partial: "account/trades/security" } %>
\ No newline at end of file
```

Reply:
And then this is the new `_security.turbo_stream.erb`:

```erb
<div class="flex items-center">
  <%= image_tag(security.logo_url, class: "rounded-full h-8 w-8 inline-block mr-2" ) %>
  <div class="flex flex-col">
    <span class="text-sm font-medium">
      <%= security.name.presence || security.ticker %>
    </span>
    <span class="text-xs text-gray-500">
      <%= "#{security.ticker} (#{security.exchange_acronym})" %>
    </span>
  </div>
</div>
```